app/assets/javascripts/govuk_publishing_components/load-analytics.js

```diff
@@ -0,0 +1,261 @@
+      'teacherservices.education.gov.uk',
+      'teaching-vacancies.service.gov.uk',
+      'to-visit-or-stay-in-the-uk.homeoffice.gov.uk',
+      'trade-tariff.service.gov.uk',
+      'tribunal-response.employmenttribunals.service.gov.uk',
+      'ukri.org',
+      'update-student-loan-employment-details.service.gov.uk',
+      'vehicle-operator-licensing.service.gov.uk',
+      'vehicleenquiry.service.gov.uk',
+      'viewdrivingrecord.service.gov.uk',
+      'view-and-prove-your-rights.homeoffice.gov.uk',
+      'view-immigration-status.service.gov.uk',
+      'visa-address-update.service.gov.uk',
+      'visas-immigration.service.gov.uk',
+      'your-defra-account.defra.gov.uk'
+    ]
+
+    // Universal Analytics variables
+    window.GOVUK.analyticsVars = window.GOVUK.analyticsVars || {}
+    window.GOVUK.analyticsVars.gaProperty = 'UA-UNSET'
+    window.GOVUK.analyticsVars.gaPropertyCrossDomain = 'UA-145652997-1'
+    window.GOVUK.analyticsVars.primaryLinkedDomains = ['account.gov.uk']
+    window.GOVUK.analyticsVars.linkedDomains = linkedDomains
+
+    if (this.arrayContains(currentDomain, this.productionDomains)) {
+      window.GOVUK.analyticsVars.gaProperty = 'UA-26179049-1'
+    } else if (this.arrayContains(currentDomain, this.stagingDomains)) {
+      window.GOVUK.analyticsVars.gaProperty = 'UA-26179049-20'
+    } else if (this.arrayContains(currentDomain, this.integrationDomains)) {
+      window.GOVUK.analyticsVars.gaProperty = 'UA-26179049-22'
+    } else {
+      window.GOVUK.analyticsVars.gaPropertyCrossDomain = 'UA-UNSET'
+    }
+
+    // Load universal analytics
+    if (typeof window.GOVUK.analyticsInit !== 'undefined') {
+      window.GOVUK.analyticsInit()
+    }
+  },
+
+  loadGa4: function (currentDomain) {
+    currentDomain = currentDomain || window.location.hostname
+    var environment = ''
+
+    // Categorise current environment and set a couple UA variables
+    if (this.arrayContains(currentDomain, this.productionDomains)) {
+      environment = 'production'
+    } else if (this.arrayContains(currentDomain, this.stagingDomains)) {
+      environment = 'staging'
+    } else if (this.arrayContains(currentDomain, this.integrationDomains)) {
+      environment = 'integration'
+    } else if (this.arrayContains(currentDomain, this.developmentDomains) || currentDomain.indexOf('.dev.gov.uk') !== -1) {
+      environment = 'development'
+    }
+
+    // If we recognise the environment (i.e. the string isn't empty), load in GA4
+    if (environment) {
+      // If analytics-ga4.js exists and our detected environment has 'initialiseGA4' set to true, load GA4.
+      if (typeof window.GOVUK.analyticsGa4.init !== 'undefined' && this.ga4EnvironmentVariables[environment].initialiseGA4) {
+        window.GOVUK.analyticsGa4 = window.GOVUK.analyticsGa4 || {}
+        window.GOVUK.analyticsGa4.vars = window.GOVUK.analyticsGa4.vars || {}
+        window.GOVUK.analyticsGa4.vars.id = this.ga4EnvironmentVariables.id
```

Comment:
If `id` is consistent across all environments, and we've already checked that the environment is valid, can the ID not just be set here, rather than fetched from the object?

Reply:
@andysellick I guess so - my thinking was it would be more readable to keep all the environment variables in one object.